### PR TITLE
Fix target callsign not resetting on TX1/TX2 slot toggle

### PR DESCRIPTION
**Pull request by patrickrb**
## Summary
- Toggling the TX time-slot (TX1 ↔ TX2) while mid-QSO left the target callsign locked on the previous station instead of resetting to CQ
- Added `shouldResetTargetOnSlotToggle()` static helper to `FT8TransmitSignal` that determines whether a slot toggle should reset the target (true when mid-QSO, false when already at CQ baseline)
- The `onToggleSlot` handler in `FT8USApp.kt` now calls `userResetToCQ()` when switching slots during an active contact, clearing target + caller queue

## Test plan
- [x] Added 3 unit tests in `FT8TransmitSignalTest` covering mid-QSO reset, CQ baseline no-reset, and edge case (order 6 with real target)
- [ ] Manual: start a QSO, toggle TX1→TX2 mid-contact, verify target callsign resets to CQ
- [ ] Manual: while calling CQ (no active target), toggle TX1→TX2, verify no unnecessary reset

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ft8cn/app/src/main/java/com/bg7yoz/ft8cn/ft8transmit/FT8TransmitSignal.java [hunk 1]
```diff
@@ -1051,6 +1051,25 @@ static boolean shouldSuppressIdleHunt(boolean huntCallsCQ, boolean autoFollowCQ,
...
+     * @param targetCallsign  the current target callsign string, if any
+     */
+    public static boolean shouldResetTargetOnSlotToggle(int functionOrder, String targetCallsign) {
+        if (functionOrder == 6
+                && (targetCallsign == null || targetCallsign.isEmpty()
+                    || "CQ".equals(targetCallsign))) {
+            return false;
+        }
+        return true;
+    }
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8cn/app/src/main/java/com/bg7yoz/ft8cn/ft8transmit/FT8TransmitSignal.java]: Fixed in 227c471. Confirmed that `setCurrentFunctionOrder()` (called by `generateFun()` inside `resetToCQ()`) does NOT post `mutableFunctionOrder`, so the LiveData can indeed be stale. Simplified `shouldResetTargetOnSlotToggle` to take only `targetCallsign`, removed the `functionOrder` parameter entirely. Now uses case-insensitive trim comparison so "CQ", "cq", "CQ " all correctly skip the reset.
